File: examples/streaming/chat_stream.py
"""
Phase 2.1 streaming demo — step events on stub path (no API key required).

Usage: python examples/streaming/chat_stream.py
"""

from __future__ import annotations

import asyncio

from arcflow import Agent, Workflow


async def main() -> None:
    wf = Workflow("chat_stream_demo")
    wf.step(Agent(name="assistant", role="helper", instructions="Reply briefly."))

    print("Streaming events:")
    event_count = 0
    async for event in wf.run_stream("Hello from ArcFlow"):
        event_count += 1
        if event.type == "token":
            print(event.text, end="", flush=True)
        elif event.type == "step_start":
            print(f"\n[step start: {event.step_id}]")
        elif event.type == "step_complete":
            print(f"[step complete: {event.step_id} in {event.duration_ms}ms]")
        else:
            print(f"[{event.type}]")
    if event_count == 0:
        raise RuntimeError("Expected at least one stream event.")
    print("\nDone.")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/streaming/chat_stream.ts
/**
 * Phase 2.1 streaming demo — step events on stub path (no API key required).
 *
 * Usage: npm run build && npx tsx examples/streaming/chat_stream.ts
 */
import { Agent, Workflow } from "../../sdk-typescript/index.js";

async function main(): Promise<void> {
  const wf = new Workflow({ name: "chat_stream_demo" });
  wf.step(
    new Agent({
      name: "assistant",
      role: "helper",
      instructions: "Reply briefly.",
    }),
  );

  process.stdout.write("Streaming events:\n");
  for await (const event of wf.runStream("Hello from ArcFlow")) {
    switch (event.type) {
      case "token":
        process.stdout.write(event.text);
        break;
      case "step_start":
        process.stdout.write(`\n[step start: ${event.step_id}]\n`);
        break;
      case "step_complete":
        process.stdout.write(
          `[step complete: ${event.step_id} in ${event.duration_ms}ms]\n`,
        );
        break;
      default:
        process.stdout.write(`[${event.type}]\n`);
    }
  }
  process.stdout.write("\nDone.\n");
}

main().catch((err: unknown) => {
  console.error(err);
  process.exit(1);
});
